iina/PlayerCore.swift

```diff
@@ -1607,10 +1607,7 @@ class PlayerCore: NSObject {
         HistoryController.shared.add(url, duration: duration.second)
       }
       if Preference.bool(for: .recordRecentFiles) && Preference.bool(for: .trackAllFilesInRecentOpenMenu) {
-        NSDocumentController.shared.noteNewRecentDocumentURL(url)
-#if DEBUG
-        DispatchQueue.main.async { (NSApp.delegate as? AppDelegate)?.saveRecentDocuments() }
-#endif
+        DispatchQueue.main.sync { (NSApp.delegate as? AppDelegate)?.noteNewRecentDocumentURL(url) }
```

Comment:
I know this isn't your code but just force unwrap this. If the application delegate is not `AppDelegate` there is no point in continuing execution

```suggestion
        DispatchQueue.main.sync { (NSApp.delegate as! AppDelegate).noteNewRecentDocumentURL(url) }
```

Reply:
Will fix.

Please do point out anything you spot that needs fixing during a review. I am always appreciative of your review comments and happy to fix anything you notice.

At the time I coded this I searched the code base and found this same pattern appears in many places. More instances use the conditional operator, but some force the conversion. I was thinking about adding an extension to `NSApp` to eliminate the duplication but decided such cleanup should wait for 1.4.0.

I am biased towards forcing such conversions. The conditional operator turns what would have been a crash seen and fixed by a developer into odd runtime behavior that can easily escape notice and end up being released to users. Troubles me that the industry views use of force operators to always be a _code smell_. So much so there was this Swift Pitch: [Moving toward deprecating force unwrap from Swift?](https://forums.swift.org/t/moving-toward-deprecating-force-unwrap-from-swift/43455) The core team decided deprecation was out of the question, responding in [this comment](https://forums.swift.org/t/moving-toward-deprecating-force-unwrap-from-swift/43455/81). But that is an example of how strongly many are against use of the force operators.

For a case like the one at hand, if I was unable to use a force operator I would use a `guard` statement that triggerd a fatal internal error if the downcast failed.